### PR TITLE
complete back port the mapping=servlet to 2.4.x

### PR DESCRIPTION
**Pull request by jfclere**
back port list:

    r1879074 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879074)
    r1879075 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879075)
    r1879076 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879076)
    r1879077 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879077)
    r1879078 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879078)
    r1879079 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879079)
    r1879080 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879080)
    r1879094 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879094)
    r1879095 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879095)
    r1879110 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879110)
    r1879111 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879111)
    r1879112 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879112)
    r1879114 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879114)
    r1879116 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879116)
    r1879117 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879117)
    r1879137 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879137)
    r1879144 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879144)
    r1879145 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879145)
    r1879147 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879147)
    r1879149 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879149)
    r1879235 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879235)
    r1879360 (https://svn.apache.org/repos/asf/httpd/httpd/trunk@1879360)

### CODE REVIEW

File: server/request.c [hunk 1]
```diff
@@ -157,24 +160,93 @@ AP_DECLARE(int) ap_some_authn_required(request_rec *r)
...
+
+    if ((access_status = ap_location_walk(r))) {
+        return access_status;
+    }
+    if ((access_status = ap_if_walk(r))) {
+        return access_status;
+    }
+
+    /* Don't set per-dir loglevel if LogLevelOverride is set */
+    if (!r->connection->log) {
```
Comment (ylavic) [inline server/request.c]: That `if (!r->connection->log)` (and comment) is from https://github.com/apache/httpd/commit/d04d01c5891a1f0f8fadd5e16f1752b915d2153f which never made it to 2.4.x (AFAICT).
For now it probably should be simply:
```
    {
        core_dir_config *d...
        ...
    }
```
in 2.4.x (with the unconditional block which aligns with trunk still).

File: include/ap_mmn.h [hunk 2]
```diff
@@ -562,14 +562,18 @@
  * 20120211.106 (2.4.49-dev) Add ap_create_request().
  * 20120211.107 (2.4.49-dev) Add ap_parse_request_line() and
  *                           ap_check_request_header()
+ * 20120211.108 (2.4.49-dev) Add ap_normalize_path(),
+ *                           pre_translate_name hook and
+ *                           Add map_encoded_one and map_encoded_all bits to
+ *                           proxy_server_conf.
```
Comment (ylavic) [inline include/ap_mmn.h:568]: Those were 3 separate MMN minor bumps in trunk, but I suppose that it's OK to merge them if they end up in a single 2.4.x backport..

File: server/request.c (hunk 1 above)
Comment (jfclere) [reply server/request.c]: OK I will fix that thanks.

Comment (jfclere) [reply server/request.c]: fixed.

**Review by ylavic — APPROVED** [review]

**Comment by ylavic** [comment]
Merged to 2.4.x in https://github.com/apache/httpd/commit/6a5d3e006b8dc8aca1a267a8607864e1c3607f61 (r1892161)

**Comment by ylavic** [comment]
Merged

**Pull request closed without merging**